src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -99,7 +99,7 @@ public abstract partial class TypeDesc
         public override int GetHashCode()
         {
             // Inherited types are expected to override
-            return RuntimeHelpers.GetHashCode(this);
+            throw new NotImplementedException();
```

Comment:
I agree with @MichalStrehovsky that would be best. However I do not understand your comment about overriding System.Object.GetHashCode, isn't it what the code is currently doing anyway?